cdap-ui/app/features/admin/controllers/namespace/templates-ctrl.js

```diff
@@ -74,52 +118,66 @@ angular.module(PKG.name + '.feature.admin')
+            vm.plugin = $filter('filter')(res, {artifact: vm.pluginConfig.artifact})[0];
+            vm.onPluginVersionSelect();
+          }
         });
     }
 
-    if (!$stateParams.pluginTemplate) {
-      // Obtaining list of plugins
-      $scope.$watch(function () { return vm.pluginType; }, function () {
-        if (!vm.pluginType) { return; }
-
-        vm.pluginName = null;
-        plugin = null;
-        vm.pluginConfig = null;
-        vm.configFetched = false;
-
-        var prom;
-        var params = {
-          pipelineType: vm.templateType,
-          namespace: $stateParams.nsadmin,
-          version: $rootScope.cdapVersion
+
+    vm.onPluginVersionSelect = function () {
+      if (!vm.plugin) { return; }
+
+      if (!vm.pluginConfig) {
+        vm.pluginConfig = {
+          _backendProperties: vm.plugin.properties,
+          properties: {},
+          lock: {}
         };
-        switch (vm.pluginType) {
-          case GLOBALS.pluginTypes[vm.templateType].source:
-            params.extensionType = GLOBALS.pluginTypes[vm.templateType].source;
-            prom = myPipelineApi.fetchSources(params).$promise;
-            break;
-          case GLOBALS.pluginTypes[vm.templateType].transform:
-            params.extensionType = GLOBALS.pluginTypes[vm.templateType].transform;
-            prom = myPipelineApi.fetchTransforms(params).$promise;
-            break;
-          case GLOBALS.pluginTypes[vm.templateType].sink:
-            params.extensionType = GLOBALS.pluginTypes[vm.templateType].sink;
-            prom = myPipelineApi.fetchSinks(params).$promise;
-            break;
-        }
-        prom.then(function (res) {
-          vm.pluginList = res;
-        });
-      });
+      } else {
+        vm.pluginConfig._backendProperties = vm.plugin.properties;
+      }
 
-      // Fetching backend properties
-      $scope.$watch(function () { return vm.pluginName; }, initialize);
-    }
+      var artifact = {
+        name: vm.plugin.artifact.name,
+        version: vm.plugin.artifact.version,
+        key: 'widgets.' + vm.plugin.name + '-' + vm.plugin.type
+      };
+
+
+      PluginConfigFactory.fetchWidgetJson(artifact.name, artifact.version, artifact.key)
+        .then(function (res) {
+          vm.configFetched = true;
+
+          vm.groupsConfig = PluginConfigFactory.generateNodeConfig(vm.pluginConfig._backendProperties, res);
+
+          angular.forEach(vm.groupsConfig.groups, function (group) {
+            angular.forEach(group.fields, function (field) {
+              if (field.defaultValue) {
+                vm.pluginConfig.properties[field.name] = vm.pluginConfig.properties[field.name] || field.defaultValue;
+              }
+            });
+          });
+
+          var configOutputSchema = vm.groupsConfig.outputSchema;
+          // If its an implicit schema, set the output schema to the implicit schema and inform ConfigActionFactory
+          if (configOutputSchema.implicitSchema) {
```

Comment:
We might be overlooking at the case where the output schema might be tied to a property in the plugin. Right now it doesn't matter (it works) as we check that in node config. But the assumption that the user has when setting the schema might not be entirely true with the way we are saving it.